src/Plugins/OrientationAnalysis/src/OrientationAnalysis/Filters/Algorithms/FindFeatureNeighborCAxisMisalignments.cpp

```diff
@@ -0,0 +1,169 @@
+Result<> FindFeatureNeighborCAxisMisalignments::operator()()
+{
+  const auto& crystalStructures = m_DataStructure.getDataRefAs<UInt32Array>(m_InputValues->CrystalStructuresArrayPath);
+  bool allPhasesHexagonal = true;
+  bool noPhasesHexagonal = true;
+  for(usize i = 1; i < crystalStructures.size(); ++i)
+  {
+    const auto crystalStructureType = crystalStructures[i];
+    const bool isHex = crystalStructureType == EbsdLib::CrystalStructure::Hexagonal_High || crystalStructureType == EbsdLib::CrystalStructure::Hexagonal_Low;
+    allPhasesHexagonal = allPhasesHexagonal && isHex;
+    noPhasesHexagonal = noPhasesHexagonal && !isHex;
+  }
+
+  if(noPhasesHexagonal)
+  {
+    return MakeErrorResult(
+        -1562, "Finding the feature neighbor c-axis mis orientation requires at least one phase to be Hexagonal-Low 6/m or Hexagonal-High 6/mmm type crystal structures but none were found.");
+  }
+
+  Result<> result;
+  if(!allPhasesHexagonal)
+  {
+    result.warnings().push_back(
+        {-1563, "Finding the feature neighbor c-axis mis orientation requires Hexagonal-Low 6/m or Hexagonal-High 6/mmm type crystal structures. Calculations for non Hexagonal phases will be "
+                "skipped and a NAN value will be used instead."});
+  }
+
+  auto& neighborList = m_DataStructure.getDataRefAs<NeighborList<int32>>(m_InputValues->NeighborListArrayPath);
+  const auto& featurePhases = m_DataStructure.getDataRefAs<Int32Array>(m_InputValues->FeaturePhasesArrayPath);
+  const auto& avgQuats = m_DataStructure.getDataRefAs<Float32Array>(m_InputValues->AvgQuatsArrayPath);
+
+  auto& cAxisMisalignmentList = m_DataStructure.getDataRefAs<NeighborList<float32>>(m_InputValues->CAxisMisalignmentListArrayName);
+  auto& avgCAxisMisalignment = m_DataStructure.getDataRefAs<Float32Array>(m_InputValues->AvgCAxisMisalignmentsArrayName);
+
+  const usize totalFeatures = featurePhases.getNumberOfTuples();
+  const usize numQuatComps = avgQuats.getNumberOfComponents();
+
+  std::vector<std::vector<float>> misalignmentLists;
+
+  float32 w = 0.0f;
+  Matrix3fR g1T;
+  g1T.fill(0.0f);
+  Matrix3fR g2T;
+  g2T.fill(0.0f);
+  Eigen::Vector3f c1{0.0f, 0.0f, 0.0f};
+  Eigen::Vector3f c2{0.0f, 0.0f, 0.0f};
+  const Eigen::Vector3f cAxis{0.0f, 0.0f, 1.0f};
+  usize hexNeighborListSize = 0;
+  uint32 phase1 = 0, phase2 = 0;
+  usize nName = 0;
+  misalignmentLists.resize(totalFeatures);
+  for(usize i = 1; i < totalFeatures; i++)
+  {
+    phase1 = crystalStructures[featurePhases[i]];
+
+    const usize quatTupleIndex1 = i * numQuatComps;
+    OrientationF oMatrix1 =
+        OrientationTransformation::qu2om<QuatF, OrientationF>({avgQuats[quatTupleIndex1], avgQuats[quatTupleIndex1 + 1], avgQuats[quatTupleIndex1 + 2], avgQuats[quatTupleIndex1 + 3]});
+
+    // transpose the g matrix so when c-axis is multiplied by it
+    // it will give the sample direction that the c-axis is along
+    g1T = OrientationMatrixToGMatrixTranspose(oMatrix1);
+    c1 = g1T * cAxis;
```

Comment:
```suggestion
    c1 = OrientationMatrixToGMatrixTranspose(oMatrix1) * cAxis;
```
make r-value to avoid excess non-trival copies and reduce overhead